### PR TITLE
backport AppDeployIntegrationTest fixes and mark stable

### PR DESCRIPTION
**Pull request by timcharper**

### CODE REVIEW

File: src/test/python/app_mock.py [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+            except:
+                logging.exception('Could not handle GET request')
+                raise
+
+        def do_POST(self):
+            try:
+                logging.debug("Got POST request")
+                return self.check_health()
+            except:
+                logging.exception('Could not handle GET request')
```
Comment (aquamatthias) [inline src/test/python/app_mock.py]: s/GET/POST

Comment (timcharper) [inline src/test/python/app_mock.py]: Good point; I'll submit a patch to fix it in master

Comment (jeschkies) [inline src/test/python/app_mock.py]: Please fix this for 1.4 as well :).

**Comment by timcharper** [comment]
@jeschkies I don't think we should fix the log line in this pull request as it is a back port. I think the optimal procedure would be to fix on master, and then back port again. Does this sound good to you?

**Comment by jeschkies** [comment]
@timcharper sure. Go ahead.

**Pull request merged**